utils/yoke/src/utils.rs

```diff
@@ -0,0 +1,25 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::Yokeable;
+use core::mem;
+
+/// This method casts `yokeable` between `&'a mut Y<'static>` and `&'a mut Y<'a>`,
+/// and passes it to `f`.
+///
+/// See [`Yokeable::transform_mut`] for why this is safe, noting that no `'static` return type
+/// can leak data from the cart or Yokeable.
+#[inline]
+pub(crate) fn transform_mut_yokeable<'a, Y, F, R>(yokeable: &'a mut Y, f: F) -> R
```

Comment:
maybe this can be defined locally inside `with_mut_return` so we don't need a utils module and the logic is all in one place

Reply:
We're planning on expanding this utils module with more of the Yokeable methods (we plan to remove them from the trait and turn them into internal helpers)

Reply:
ok